Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs

```diff
@@ -159,16 +159,22 @@ CASE Coalesce(F.RDB$FIELD_TYPE, 0)
 				WHEN 13 THEN 'TIME'
 				WHEN 14 THEN 'CHAR(' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ') '
 				WHEN 16 THEN
-				CASE F.RDB$FIELD_SUB_TYPE
-				 WHEN 0 THEN 'BIGINT'
-				 ELSE 'DECIMAL'
-				END
+				 CASE F.RDB$FIELD_SUB_TYPE
+				  WHEN 0 THEN 'BIGINT'
+				  ELSE 'DECIMAL'
+				 END
+                WHEN 23 THEN 'BOOLEAN'
 				WHEN 27 THEN 'DOUBLE PRECISION'
 				WHEN 35 THEN 'TIMESTAMP'
 				WHEN 37 THEN 'VARCHAR(' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ')'
 				WHEN 40 THEN 'CSTRING' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ')'
 				WHEN 45 THEN 'BLOB_ID'
-				WHEN 261 THEN 'BLOB SUB_TYPE ' || F.RDB$FIELD_SUB_TYPE
+				WHEN 261 THEN 'BLOB SUB_TYPE ' ||
+				  CASE F.RDB$FIELD_SUB_TYPE
+				    WHEN 0 THEN 'BINARY'
+				    WHEN 1 THEN 'TEXT'
+				    ELSE F.RDB$FIELD_SUB_TYPE
+          END
```

Comment:
Nit: The `END` has wrong indentation.

Reply:
This one is wrong on my side aswell, will fix